### PR TITLE
Improve rules for Docker listener

### PR DESCRIPTION
**Pull request by vikman90**
## Issues detected

1. `docker exec` incorrectly triggered rules 87901 (container created) and 87903 (container started).
2. Alerts from Docker listener included the _full_log_ even through the input is fully decoded (JSON).

### Point (2) rationale:

`docker exec` produces a log like this:

```json
{
  "docker": {
    "status": "exec_start: bash ",
    (...)
    "id": "7697fa3fec2a0221e448886e5c574c082309fbe0cac8b358d11c3830164c5e37"
  },
  "integration": "docker"
}
```

Rule 87901 contains this filter:
```xml
<field name="docker.status">start</field>
```

This makes `docker exec` trigger rule 87903.

## Changes proposed

1. Disable the _full_log_ in the Docker listener rules.
2. Fix `<field>` filters in those rules to match the exact string (instead of a substring).
3. Add a specific rule (87907) for command execution in containers (`docker exec`).

## Sample alert

```json
{
  "timestamp": "2019-02-21T18:01:44.193-0800",
  "rule": {
    "level": 3,
    "description": "Command run in container reverent_ishizaka",
    "id": "87907",
    "firedtimes": 2,
    "mail": false,
    "groups": [
      "docker"
    ]
  },
  "agent": {
    "id": "001",
    "name": "DemoAgent",
    "ip": "192.168.33.1"
  },
  "manager": {
    "name": "stretch64"
  },
  "id": "1550800904.201466",
  "decoder": {
    "name": "json"
  },
  "data": {
    "docker": {
      "status": "exec_start: echo Hello World",
      "timeNano": "1550800904187913984.000000",
      "from": "ubuntu",
      "Actor": {
        "Attributes": {
          "execID": "4270b190f78fcad1849ce23300b1be4e1907111747457ee470e878214aeec38c",
          "image": "ubuntu",
          "name": "reverent_ishizaka"
        },
        "ID": "4b478281f9c8a9d504a1835b409e4c35f8dd95cff1d21da0dd440effd9d85fe3"
      },
      "time": "1550800904",
      "Action": "exec_start: echo Hello World",
      "scope": "local",
      "Type": "container",
      "id": "4b478281f9c8a9d504a1835b409e4c35f8dd95cff1d21da0dd440effd9d85fe3"
    },
    "integration": "docker"
  },
  "location": "Wazuh-Docker"
}
```

### CODE REVIEW

**Comment by vikman90** [comment]
Waiting for further changes by @cristgl.

**Comment by cristgl** [comment]
At the time, every Docker event generated matches a rule and its specific alert is triggered. This seems like a problem, as each command can generate more than one event.

I have written a comment at the issue https://github.com/wazuh/wazuh-ruleset/issues/294 where I relate the Docker command with the rule matched, the event and the alert triggered. There, it can be seen that almost every command generates more than one event, this means that, for example, for the event `docker stop`, events for status `kill`, `die`, `stop` and `destroy` are shown, which means 4 alerts.

File: rules/0560-docker_integration_rules.xml [hunk 1]
```diff
@@ -13,44 +13,176 @@ ID: 87900 - 87999
...
+    <rule id="87907" level="3">
+        <if_sid>87900</if_sid>
+        <field name="docker.status">^exec_start: </field>
+        <description>Command run in container $(docker.Actor.Attributes.name). Action: '$(docker.Action)'</description>
+        <options>no_full_log</options>
+    </rule>
+
+    <rule id="87908" level="3">
+        <if_sid>87907</if_sid>
+        <field name="docker.status">^exec_start: bash</field>
```
Comment (vikman90) [inline rules/0560-docker_integration_rules.xml]: This condition does only catch:
- [x] _bash_
- [ ] _/bin/bash_
- [ ] _sh_
- [ ] _/bin/sh_
- [ ] _dash_
- [ ] _/bin/dash_

File: rules/0560-docker_integration_rules.xml [hunk 2]
```diff
@@ -13,44 +13,176 @@ ID: 87900 - 87999
...
+    </rule>
+
+    <rule id="87917" level="3">
+        <if_sid>87900</if_sid>
+        <field name="docker.status">^delete$</field>
+        <description>Container $(docker.Actor.Attributes.name) deleted</description>
+        <options>no_full_log</options>
+    </rule>
+
+    <rule id="87918" level="3">
```
Comment (vikman90) [inline rules/0560-docker_integration_rules.xml]: Is this rule a duplicate of _87907_?

**Comment by chemamartinez** [comment]
Why don't we add a rule for the `pull` action? Here we can see the event generated by the integration:

```json
{
  "integration": "docker",
  "docker": {
    "status": "pull",
    "id": "centos:latest",
    "Type": "image",
    "Action": "pull",
    "Actor": {
      "ID": "centos:latest",
      "Attributes": {
        "name": "centos",
        "org.label-schema.build-date": "20181205",
        "org.label-schema.license": "GPLv2",
        "org.label-schema.name": "CentOS Base Image",
        "org.label-schema.schema-version": "1.0",
        "org.label-schema.vendor": "CentOS"
      }
    },
    "scope": "local",
    "time": 1551182693,
    "timeNano": 1551182693249997000
  }
}
```

File: rules/0560-docker_integration_rules.xml (hunk 2 above)
Comment (cristgl) [reply rules/0560-docker_integration_rules.xml]: Fixed, now rule 87917 checks if an image is untagged.

**Pull request merged**